akka-typed/src/main/scala/akka/typed/ActorSystem.scala

```diff
@@ -146,9 +148,10 @@ abstract class ActorSystem[-T] extends ActorRef[T] with Extensions { this: inter
   def systemActorOf[U](behavior: Behavior[U], name: String, props: Props = Props.empty)(implicit timeout: Timeout): Future[ActorRef[U]]
 
   /**
-   * Return a reference to this system’s [[akka.typed.patterns.Receptionist$]].
+   * Return a reference to this system’s [[akka.typed.receptionist.Receptionist]].
    */
-  def receptionist: ActorRef[patterns.Receptionist.Command]
+  def receptionist: ActorRef[Receptionist.Command] =
+    Receptionist(this).receptionist
```

Comment:
It seems extensions are like a good pattern in general instead of putting everything into the actor system directly. I see this method only as a convenience shortcut and the extension as the real thing. But that's debatable, of course.

Reply:
I think I reacted because I don't like cycles in code, the actor system has a receptionist which is there because there is an extension which is there because the actor system loads it. Not super strong feelings about it, just makes me itch a bit :)